examples/calendar/frontend/src/Profile/Profile.tsx

```diff
@@ -0,0 +1,196 @@
+import "./Profile.css";
+
+import jwtDecode from "jwt-decode";
+import React, { useState, useEffect, startTransition } from "react";
+
+import { Auth, User } from "../types";
+import axios from "axios";
+import Calendar from 'react-calendar';
+import 'react-calendar/dist/Calendar.css';
+import {ToggleButton, Button} from 'react-bootstrap';
+
+interface Props {
+	auth: Auth;
+	onLoggedOut: () => void;
+}
+
+interface Time {
+	startTime: Date;
+	checked: boolean;
+	disabled: boolean;
+}
+interface SetBooking {
+	timeStarts: number[];
+	timeEnds: number[];
+	status: string;
+	userId: number;
+}
+interface State {
+	loading: boolean;
+	user?: {
+		id: number;
+		username: string;
+	};
+	username: string;
+	times?: Time[]
+}
+
+interface JwtDecoded {
+	payload: {
+		id: string;
+		publicAddress: string;
+	};
+}
+
+export const Profile = ({ auth, onLoggedOut }: Props): JSX.Element => {
+
+	const getTimeAvailAPI = async (userId: string) => {
+		const fetchAvailTimes = (await axios.get(`${process.env.REACT_APP_BACKEND_URL}/time/${userId}`, {
+			headers: {
+				Authorization: `Bearer ${accessToken}`
+			}
+		})).data;
+		return fetchAvailTimes;
+	}
+	const createAppointmentAPI = async () => {
+		const createAppointment = (await axios.post(`${process.env.REACT_APP_BACKEND_URL}/appointments`, {} ,{
+			headers: {
+				Authorization: `Bearer ${accessToken}`
+			}
+		})).data;
+		return createAppointment;
+	}
+	const setTimeAvailAPI = async (availableTimes: SetBooking) => {
+		const setAvailTimes = (await axios.post(`${process.env.REACT_APP_BACKEND_URL}/time`, {availableTimes} ,{
+			headers: {
+				'Content-Type': 'application/json',
+				Authorization: `Bearer ${accessToken}`
+			}
+		})).data;
+		return setAvailTimes;
+	}
+	const timesInitial = () => {
+		const timeReturn = [];
+		var someDate = new Date(selectedDate);
+		someDate.setHours(8, 0, 0);
```

Comment:
please add a JSDoc as to what is happening here and at line 78

Reply:
@vaibhavgeek

Reply:
It gets the Time availability API. I feel the code is self explanatory here.